spring-boot-admin-server-ui/src/main/frontend/views/instances/gateway/routes.vue

```diff
@@ -0,0 +1,246 @@
+      || route.route_definition.filters.filter(f => Object.values(f.args).filter(av => av.toLowerCase().includes(keyword)).length > 0).length > 0;
+  };
+
+  const sortRoutes = (globalFilters, sort) => {
+    if (sort === 'route_id') {
+      return globalFilters.slice().sort(function (a, b) {
+        return a.route_id.localeCompare(b.route_id)
+      })
+    } else if (sort === 'order') {
+      return globalFilters.slice().sort(function (a, b) {
+        return a.order - b.order
+      })
+    }
+
+    return globalFilters;
+  };
+
+  export default {
+    components: {routeDetailControl},
+    props: {
+      instance: {
+        type: Instance,
+        required: true
+      }
+    },
+    data: () => ({
+      hasLoaded: false,
+      error: null,
+      routesData: null,
+      routesFilter: null,
+      sort: 'undefined',
+      showDetails: {},
+      clearRoutesCacheStatus: null
+    }),
+    computed: {
+      routes() {
+        if (!this.routesData) {
+          return [];
+        }
+        if (!this.routesFilter) {
+          return sortRoutes(this.routesData, this.sort);
+        }
+        return sortRoutes(this.routesData.filter(route => !this.routesFilter || filterRoutesByKeyword(route, this.routesFilter.toLowerCase())), this.sort);
+      }
+    },
+    created() {
+      this.fetchRoutesData();
+    },
+    methods: {
+      async fetchRoutesData() {
+        this.error = null;
+        try {
+          const res = await this.instance.fetchRoutesData();
+          this.routesData = uniqBy(res.data, 'route_id');
+        } catch (error) {
+          console.warn('Fetching routes failed:', error);
+          this.error = {action:'Fetching gateway routes failed:', value:error};
+        }
+        this.hasLoaded = true;
+      },
+      getFilterFn() {
+        if (!this.filter || this.filter === '') {
+          return () => true;
+        }
+        const regex = new RegExp(this.filter, 'i');
+        return route => (route.route_id.match(regex));
+      },
+      async deleteRoute(dialog) {
+        let button = dialog.node;
+        let routeId = button.dataset.route_id;
+
+        try {
+          await this.instance.deleteRoute(routeId);
+          setTimeout(() => this.$emit('route-deleted'), 2500);
+        } catch (response) {
+          console.warn('Deleting route failed:', response);
+          this.error = {
+            action:'Deleting route ' + routeId + ' failed:', 
+            value:response};
+        }
+
+        dialog.close();
+      },
+      clearRoutesCache() {
+        const vm = this;
+        from(vm.instance.clearRoutesCache())
+          .pipe(listen(status => vm.clearRoutesCacheStatus = status))
+          .subscribe({
+            complete: () => {
+              setTimeout(() => vm.clearRoutesCacheStatus = null, 2500);
+              return vm.$emit('reset');
+            },
+            error: () => vm.$emit('reset')
+        });
+
+        try {
+          this.instance.clearRoutesCache();
```

Comment:
Isn't `this.instance.clearRoutesCache();`invoked here a second time? why is it necessary?